libnestutil/type_traits.h

```diff
@@ -102,7 +102,7 @@ template < class T >
 struct remove_pointer;
 template < class T, class U >
 struct is_same;
-#if !defined( _MSC_VER ) && !( defined( __GNUC__ ) && __GNUC__ <= 3 )
+#if not defined( _MSC_VER ) && not( defined( __GNUC__ ) && __GNUC__ <= 3 )
```

Comment:
'not' isn't supported by the preprocessor language. Since 'not' is syntactically correct (it is a valid macro name), the preprocessor will generate a warning in the best case. 'not' should be replaced with '!' or the expression can also be re-written using #ifndef.